# Leqi Zhong final project notebook 3

Notebook 3 includes data visualization for year 2122 data.

In [1]:
import pandas as pd
import altair as alt

In [2]:
data_2122_clean = pd.read_csv('data_2122_clean.csv')
data_2122_clean.head()

Academic Year County Name                       District Name  \
0     2021-2022     Alameda  Alameda County Office of Education   
1     2021-2022     Alameda  Alameda County Office of Education   
2     2021-2022     Alameda  Alameda County Office of Education   
3     2021-2022     Alameda  Alameda County Office of Education   
4     2021-2022     Alameda  Alameda County Office of Education   

                               School Name                   School Type  \
0                 Alameda County Community              County Community   
1       Alameda County Juvenile Hall/Court        Juvenile Court Schools   
2                   Alternatives in Action         High Schools (Public)   
3                Aurum Preparatory Academy  Junior High Schools (Public)   
4  Community School for Creative Education   Elementary Schools (Public)   

          Educational_Type Low Grade High Grade  2122_Enrollment  \
0  County Community School         K         12               57   
1    Juvenile Court School         K         12               64   
2              Traditional         9         12              150   
3              Traditional         6          8              164   
4              Traditional         K          8              202   

   2122_Free_Meal_Count  2122_Free_Meal_Percent  2122_FRPM_Count  \
0                    45                   0.789               47   
1                    64                   1.000               64   
2                   113                   0.753              134   
3                   135                   0.823              148   
4                   162                   0.802              177   

   2122_FRPM_Percent  
0              0.825  
1              1.000  
2              0.893  
3              0.902  
4              0.876

Firstly, let's see free meals count devided by counties.

In [3]:
data_2122_clean['County Name'].value_counts()

Los Angeles        2279
San Diego           796
Orange              655
San Bernardino      584
Riverside           525
Santa Clara         449
Sacramento          397
Alameda             387
Fresno              357
Contra Costa        294
Kern                272
San Joaquin         247
Ventura             228
Stanislaus          207
Sonoma              200
San Mateo           199
Tulare              194
Placer              143
Monterey            143
San Francisco       129
Santa Barbara       126
Merced              109
Solano              105
Shasta              102
Humboldt             89
Marin                87
San Luis Obispo      87
Butte                87
Santa Cruz           87
Madera               78
El Dorado            74
Imperial             71
Yolo                 68
Kings                66
Mendocino            66
Sutter               46
Yuba                 45
Napa                 42
Tehama               41
Siskiyou             41
Nevada               38
Lake            

I'd like to clean the data frame a bit, remove columns that I don't need.

In [5]:
freemeals_by_county_2122 = data_2122_clean [['County Name', '2122_Free_Meal_Percent']].copy()
freemeals_by_county_2122

County Name  2122_Free_Meal_Percent
0         Alameda                   0.789
1         Alameda                   1.000
2         Alameda                   0.753
3         Alameda                   0.823
4         Alameda                   0.802
...           ...                     ...
10553        Yuba                   1.000
10554        Yuba                   0.711
10555        Yuba                   0.000
10556        Yuba                   0.600
10557        Yuba                   0.429

[10558 rows x 2 columns]

In [6]:
#alt.Chart(freemeals_by_county_2122).mark_bar().encode(
#    x='County Name',
#    y='Free_Meal_Percent'
#)

An error says the rows are greater than 5000. Maybe we can narrow down the data then make a chart.

## LA School free meal count

I'm filtering out LA's schools data.

In [7]:
LA_schools_2122 = data_2122_clean[
    (data_2122_clean['County Name'] == 'Los Angeles')
].copy()
LA_schools_2122

Academic Year  County Name                           District Name  \
1885     2021-2022  Los Angeles  Los Angeles County Office of Education   
1886     2021-2022  Los Angeles  Los Angeles County Office of Education   
1887     2021-2022  Los Angeles  Los Angeles County Office of Education   
1888     2021-2022  Los Angeles  Los Angeles County Office of Education   
1889     2021-2022  Los Angeles  Los Angeles County Office of Education   
...            ...          ...                                     ...   
4159     2021-2022  Los Angeles                        Wiseburn Unified   
4160     2021-2022  Los Angeles                        Wiseburn Unified   
4161     2021-2022  Los Angeles           SBE - Academia Avance Charter   
4162     2021-2022  Los Angeles  SBE - Los Angeles College Prep Academy   
4163     2021-2022  Los Angeles          SBE - Eagle Collegiate Academy   

                                      School Name  \
1885                       Afflerbaugh-Paige Camp   
1886                           Alma Fuerte Public   
1887         Animo City of Champions Charter High   
1888            Aspire Antonio Maria Lugo Academy   
1889  Aspire Ollin University Preparatory Academy   
...                                           ...   
4159                    Richard Henry Dana Middle   
4160                      Success Learning Center   
4161                      Academia Avance Charter   
4162             Los Angeles College Prep Academy   
4163                     Eagle Collegiate Academy   

                                    School Type          Educational_Type  \
1885                     Juvenile Court Schools     Juvenile Court School   
1886                Elementary Schools (Public)               Traditional   
1887                      High Schools (Public)               Traditional   
1888                Elementary Schools (Public)               Traditional   
1889                      K-12 Schools (Public)               Traditional   
...                                         ...                       ...   
4159       Intermediate/Middle Schools (Public)               Traditional   
4160         Special Education Schools (Public)  Special Education School   
4161                      K-12 Schools (Public)               Traditional   
4162                      High Schools (Public)               Traditional   
4163  Elemen Schools In 1 School Dist. (Public)               Traditional   

     Low Grade High Grade  2122_Enrollment  2122_Free_Meal_Count  \
1885         7         12               29                    29   
1886         K          8              123                    84   
1887         9         12              343                   263   
1888         K          6              410                   305   
1889         6         12              559                   475   
...        ...        ...              ...                   ...   
4159         6          8              956                   271   
4160         K         12               21                     9   
4161         6         12              263                   207   
4162         9         12               47                    23   
4163         K          8               12                     5   

      2122_Free_Meal_Percent  2122_FRPM_Count  2122_FRPM_Percent  
1885                   1.000               29              1.000  
1886                   0.683               91              0.740  
1887                   0.767              298              0.869  
1888                   0.744              357              0.871  
1889                   0.850              502              0.898  
...                      ...              ...                ...  
4159                   0.283              366              0.383  
4160                   0.429               11              0.524  
4161                   0.787              221              0.840  
4162                   0.489               23          

In [9]:
alt.Chart(LA_schools_2122).mark_bar().encode(
    x='District Name',
    y='2122_Free_Meal_Percent'
).properties(
    title={
        'text': 'Los Angeles Schools',
        'subtitle' : ['Percentage of free meals by school districts']
    }
)

alt.Chart(...)

## LA School free meals by type of School

In [10]:
alt.Chart(LA_schools_2122).mark_bar().encode(
    x='School Type',
    y='2122_Free_Meal_Percent'
).properties(
    title={
        'text': 'Los Angeles Schools',
        'subtitle' : ['Percentage of free meals by school type']
    }
)

alt.Chart(...)

## LA traditional high school free meal percent

First, let's pull all the traditional school data.

In [11]:
traditional_schools_2122 = data_2122_clean[
    (data_2122_clean['Educational_Type'] == 'Traditional')
].copy()
traditional_schools_2122

Academic Year County Name                       District Name  \
2         2021-2022     Alameda  Alameda County Office of Education   
3         2021-2022     Alameda  Alameda County Office of Education   
4         2021-2022     Alameda  Alameda County Office of Education   
5         2021-2022     Alameda  Alameda County Office of Education   
6         2021-2022     Alameda  Alameda County Office of Education   
...             ...         ...                                 ...   
10549     2021-2022        Yuba                           Wheatland   
10550     2021-2022        Yuba                           Wheatland   
10551     2021-2022        Yuba                           Wheatland   
10552     2021-2022        Yuba                           Wheatland   
10557     2021-2022        Yuba                Wheatland Union High   

                                   School Name  \
2                       Alternatives in Action   
3                    Aurum Preparatory Academy   
4      Community School for Creative Education   
5         Connecting Waters Charter - East Bay   
6                                  Cox Academy   
...                                        ...   
10549                               Bear River   
10550                     Lone Tree Elementary   
10551                Wheatland Charter Academy   
10552                     Wheatland Elementary   
10557                     Wheatland Union High   

                                School Type Educational_Type Low Grade  \
2                     High Schools (Public)      Traditional         9   
3              Junior High Schools (Public)      Traditional         6   
4               Elementary Schools (Public)      Traditional         K   
5                     K-12 Schools (Public)      Traditional         K   
6               Elementary Schools (Public)      Traditional         P   
...                                     ...              ...       ...   
10549  Intermediate/Middle Schools (Public)      Traditional         4   
10550           Elementary Schools (Public)      Traditional         K   
10551           Elementary Schools (Public)      Traditional         K   
10552           Elementary Schools (Public)      Traditional         K   
10557                 High Schools (Public)      Traditional         9   

      High Grade  2122_Enrollment  2122_Free_Meal_Count  \
2             12              150                   113   
3              8              164                   135   
4              8              202                   162   
5             12              514                   107   
6              5              516                   321   
...          ...              ...                   ...   
10549          8              568                   228   
10550          5              351                    68   
10551          5               87                    14   
10552          3              340                   178   
10557         12             1012                   434   

       2122_Free_Meal_Percent  2122_FRPM_Count  2122_FRPM_Percent  
2                       0.753              134              0.893  
3                       0.823              148              0.902  
4                       0.802              177              0.876  
5                       0.208              124              0.241  
6                       0.622              339              0.657  
...                       ...              ...                ...  
10549                   0.401              273              0.481  
10550                   0.194              119              0.339  
10551                   0.161               26              0.299  
10552                   0.524              189              0.556  
10557                   0.429              750              0.741  

[8854 rows x 13 columns]

In [13]:
traditional_schools_2122.sort_values(by=['2122_Free_Meal_Percent'], ascending=False)

Academic Year  County Name                   District Name  \
9362      2021-2022     Siskiyou                Seiad Elementary   
7490      2021-2022    San Diego               San Diego Unified   
466       2021-2022        Butte        Palermo Union Elementary   
9356      2021-2022     Siskiyou  Klamath River Union Elementary   
10056     2021-2022       Tulare             Rockford Elementary   
...             ...          ...                             ...   
9222      2021-2022   Santa Cruz         Soquel Union Elementary   
4524      2021-2022        Modoc             Modoc Joint Unified   
4257      2021-2022        Marin         Laguna Joint Elementary   
1129      2021-2022       Fresno      Kings Canyon Joint Unified   
3551      2021-2022  Los Angeles                Monrovia Unified   

                                             School Name  \
9362                                    Seiad Elementary   
7490                    City Heights Preparatory Charter   
466                                               Honcut   
9356                            Klamath River Elementary   
10056                                Rockford Elementary   
...                                                  ...   
9222                                         Opal Cliffs   
4524                               State Line Elementary   
4257                                       Lincoln Union   
1129   Kings Canyon Unified School District Wide Pres...   
3551                        Canyon Early Learning Center   

                                     School Type Educational_Type Low Grade  \
9362   Elemen Schools In 1 School Dist. (Public)      Traditional         K   
7490                       K-12 Schools (Public)      Traditional         6   
466                  Elementary Schools (Public)      Traditional         K   
9356   Elemen Schools In 1 School Dist. (Public)      Traditional         K   
10056  Elemen Schools In 1 School Dist. (Public)      Traditional         K   
...                                          ...              ...       ...   
9222                 Elementary Schools (Public)      Traditional         P   
4524                 Elementary Schools (Public)      Traditional         K   
4257   Elemen Schools In 1 School Dist. (Public)      Traditional         K   
1129                                   Preschool      Traditional         P   
3551                                   Preschool      Traditional         P   

      High Grade  2122_Enrollment  2122_Free_Meal_Count  \
9362           8               11                    11   
7490          12              123                   123   
466            3               14                    14   
9356           8                3                     3   
10056          8              323                   323   
...          ...              ...                   ...   
9222           8               16                     0   
4524           5                4                     0   
4257           6                7                     0   
1129           P                1                     0   
3551           P                1                     0   

       2122_Free_Meal_Percent  2122_FRPM_Count  2122_FRPM_Percent  
9362                      1.0               11                1.0  
7490                      1.0              123                1.0  
466                       1.0               14                1.0  
9356                      1.0                3                1.0  
10056                     1.0              323                1.0  
...                       ...              ...                ...  
9222                      0.0                0                0.0  
4524                      0.0                0                0.0  
4257                      0.0                0                0.0  
1129                      0.0                0                0.0  
3551                      0.0                0                

There're 8854 rows, let's narrow it down a bit.

In [14]:
traditional_schools_2122['School Type'].value_counts()

Elementary Schools (Public)                  5648
High Schools (Public)                        1291
Intermediate/Middle Schools (Public)         1258
K-12 Schools (Public)                         368
Elemen Schools In 1 School Dist. (Public)     224
Junior High Schools (Public)                   50
Preschool                                       9
High Schools In 1 School Dist. (Public)         6
Name: School Type, dtype: int64

Okay, I would like to make charts about middle and high schools.

In [15]:
traditional_high_schools_2122 = traditional_schools_2122[
    (traditional_schools_2122['School Type'] == 'High Schools (Public)') |
    (traditional_schools_2122['School Type'] == 'Junior High Schools (Public)') |
    (traditional_schools_2122['School Type'] == 'Intermediate/Middle Schools (Public)')
].copy()
traditional_high_schools_2122

Academic Year County Name                       District Name  \
2         2021-2022     Alameda  Alameda County Office of Education   
3         2021-2022     Alameda  Alameda County Office of Education   
10        2021-2022     Alameda  Alameda County Office of Education   
17        2021-2022     Alameda                     Alameda Unified   
26        2021-2022     Alameda                     Alameda Unified   
...             ...         ...                                 ...   
10538     2021-2022        Yuba            Marysville Joint Unified   
10544     2021-2022        Yuba            Marysville Joint Unified   
10548     2021-2022        Yuba              Plumas Lake Elementary   
10549     2021-2022        Yuba                           Wheatland   
10557     2021-2022        Yuba                Wheatland Union High   

                          School Name                           School Type  \
2              Alternatives in Action                 High Schools (Public)   
3           Aurum Preparatory Academy          Junior High Schools (Public)   
10               Oakland Unity Middle  Intermediate/Middle Schools (Public)   
17                       Alameda High                 High Schools (Public)   
26                     Lincoln Middle  Intermediate/Middle Schools (Public)   
...                               ...                                   ...   
10538                 Marysville High                 High Schools (Public)   
10544       Yuba Gardens Intermediate          Junior High Schools (Public)   
10548  Riverside Meadows Intermediate  Intermediate/Middle Schools (Public)   
10549                      Bear River  Intermediate/Middle Schools (Public)   
10557            Wheatland Union High                 High Schools (Public)   

      Educational_Type Low Grade High Grade  2122_Enrollment  \
2          Traditional         9         12              150   
3          Traditional         6          8              164   
10         Traditional         6          8              108   
17         Traditional         9         12             1808   
26         Traditional         6          8              819   
...                ...       ...        ...              ...   
10538      Traditional         9         12             1024   
10544      Traditional         7          8              762   
10548      Traditional         6          8              440   
10549      Traditional         4          8              568   
10557      Traditional         9         12             1012   

       2122_Free_Meal_Count  2122_Free_Meal_Percent  2122_FRPM_Count  \
2                       113                   0.753              134   
3                       135                   0.823              148   
10                       84                   0.778               90   
17                      311                   0.172              327   
26                      110                   0.134              115   
...                     ...                     ...              ...   
10538                   554                   0.541              589   
10544                   571                   0.749              605   
10548                   124                   0.282              170   
10549                   228                   0.401              273   
10557                   434                   0.429              750   

       2122_FRPM_Percent  
2                  0.893  
3                  0.902  
10                 0.833  
17                 0.181  
26                 0.140  
...                  ...  
10538              0.575  
10544              0.794  
10548              0.386  
10549              0.481  
10557              0.741  

[2599 rows x 13 columns]

In [17]:
alt.Chart(traditional_high_schools_2122).mark_bar().encode(
    x='County Name',
    y='2122_Free_Meal_Percent'
).properties(
    title={
        'text': 'Traditional high schools',
        'subtitle' : ['Percentage of free meals by county']
    }
)

alt.Chart(...)